skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -0,0 +1,192 @@
+            }
+        });
+    }
+
+    /**
+     * Creates a subscription for listening to all hotspot events being send through the
+     * application's {@link RxEventBus}
+     */
+    private Disposable addBluetoothEventSubscription() {
+        return rxEventBus.register(BluetoothEvent.class)
+                .subscribeOn(schedulerProvider.io())
+                .observeOn(schedulerProvider.androidThread())
+                .subscribe(bluetoothEvent -> {
+                    switch (bluetoothEvent.getStatus()) {
+                        case CONNECTED:
+                            progressDialog.setMessage("Connected, start downloading...");
+                            isConnected = true;
+                            break;
+                        case DISCONNECTED:
+                            isConnected = false;
+                            break;
+                    }
+                });
+    }
+
+    private Disposable addDownloadEventSubscription() {
+        return rxEventBus.register(DownloadEvent.class)
+                .subscribeOn(schedulerProvider.io())
+                .observeOn(schedulerProvider.androidThread())
+                .subscribe(downloadEvent -> {
+                    switch (downloadEvent.getStatus()) {
+                        case QUEUED:
+                            Toast.makeText(this, R.string.download_queued, Toast.LENGTH_SHORT).show();
+                            break;
+                        case DOWNLOADING:
+                            int progress = downloadEvent.getCurrentProgress();
+                            int total = downloadEvent.getTotalSize();
+                            String alertMsg = getString(R.string.receiving_items, String.valueOf(progress), String.valueOf(total));
+                            progressDialog.setTitle(getString(R.string.receiving_title));
+                            progressDialog.setMessage(alertMsg);
+                            break;
+                        case FINISHED:
+                            progressDialog.dismiss();
+                            Toast.makeText(this, getString(R.string.tv_form_send_success), Toast.LENGTH_SHORT).show();
+                            break;
+                        case ERROR:
+                            progressDialog.dismiss();
+                            Toast.makeText(this, getString(R.string.error_while_downloading, downloadEvent.getResult()), Toast.LENGTH_SHORT).show();
+                            break;
+                        case CANCELLED:
+                            progressDialog.dismiss();
+                            Toast.makeText(this, getString(R.string.canceled), Toast.LENGTH_LONG).show();
+                            break;
+                    }
+                }, Timber::e);
+    }
+
+    @Override
+    public void foundDevice(BluetoothDevice device) {
+        bluetoothListAdapter.addDevice(device);
+    }
+
+    @Override
+    public void onItemClick(BluetoothDevice dev) {
```

Comment:
Please use `device` instead of `dev` for better readability

Reply:
improved.